packages/manager/tools/dev-server-config/src/sso/index.js

```diff
@@ -0,0 +1,157 @@
+/* eslint-disable no-console */
+const base64url = require('base64url');
+const cookie = require('cookie');
+const proxy = require('request');
+
+const CONFIG = {
+  ssoAuth: {
+    eu: {
+      host: 'www.ovh.com',
+      baseUrl: 'https://www.ovh.com/cgi-bin/crosslogin.cgi',
+      devLoginUrl: 'https://www.ovh.com/auth/requestDevLogin/',
+    },
+    ca: {
+      host: 'ca.ovh.com',
+      devLoginUrl: 'https://ca.ovh.com/auth/requestDevLogin/',
+      baseUrl: 'https://ca.ovh.com/cgi-bin/crosslogin.cgi',
+    },
+    us: {
+      host: 'us.ovhcloud.com',
+      devLoginUrl: 'https://us.ovhcloud.com/auth/requestDevLogin/',
+      baseUrl: 'https://us.ovhcloud.com/cgi-bin/crosslogin.cgi',
+    },
+  },
+};
+
+module.exports = class Sso {
+  config;
+
+  constructor(region) {
+    this.config = CONFIG.ssoAuth[region];
+  }
+
+  login(req, res) {
+    console.log('[SSO] - crosslogin');
+
+    const { headers } = req;
+    headers.host = this.config.host;
+
+    const location = new Promise((resolve) => {
+      proxy.get(
+        {
+          url: this.config.baseUrl + req.url,
+          headers,
+          proxy: this.config.proxy ? this.config.proxy.host : '',
+          followRedirect: false,
+        },
+        (err, resp) => {
+          if (err) {
+            console.error('[SSO] - crosslogin - error: ', err);
+            return resp.status(500);
+          }
+
+          const cookies = resp.headers['set-cookie'];
+
+          cookies.reverse().forEach((someCookie) => {
+            const parsedCookie = cookie.parse(someCookie);
+
+            if (parsedCookie['CA.OVH.SES']) {
```

Comment:
Is this logic found that was written another way ?

Reply:
Come from https://github.com/ovh/manager/blob/master/packages/manager/tools/webpack-dev-server/src/sso/index.ts#L63-L75